### PR TITLE
Rename CollectionCreationExpression to CollectionExpression

### PR DESCRIPTION
**Pull request by cston**
See https://github.com/dotnet/roslyn/issues/68687#issuecomment-1603314314

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs
Comment (333fred) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:1]: There are a number of instances of the text "collection literal" in this file, probably worth renaming those as well.

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IArrayElementReferenceExpression.cs [hunk 1]
```diff
@@ -532,7 +532,7 @@ public void F()
                 Diagnostic(ErrorCode.ERR_CollectionLiteralNoTargetType, "[0]").WithLocation(6, 27)
             };
 
-            VerifyOperationTreeAndDiagnosticsForTest<CollectionCreationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+            VerifyOperationTreeAndDiagnosticsForTest<CollectionExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
```
Comment (333fred) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IArrayElementReferenceExpression.cs:535]: We should make sure to reword errors after LDM decides whether we want to market these things as literals or expressions.

File: src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs:1]: Nit: should we rename the file?

**Review by 333fred — APPROVED** [review]
Couple of minor comments about things still saying or named "collection literal"

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs
Comment (cston) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:1]: There are a number of instances of "collection literal" in other files as well. I will leave those as-is for now, and we can change those, along with error messages, if we decide to make a broader rename later.

**Pull request merged**